### DISCUSSION TITLE
Xll Addin

### DISCUSSION THREAD

**Discussion by hxhgxy** [Excel-DNA/ExcelDna#630] [Q&A]
I tried to install Xll addin on my friend's computer, there is no error message, but there is no ribbon installed.
I can install VSTO addin on his computer.

What shall I have a check?

Thanks a lot!

**Comment by govert** [comment]
Do you have functions in the add-in, and are they working?
You can check under **File -> Options -> Add-Ins**, then at the bottom: **Manage: Disabled Items**. See if your add-in appears there. The add-in gets disabled if it causes errors in Excel, which can happen if there are some dependencies that can't be loaded.
![image](https://github.com/Excel-DNA/ExcelDna/assets/414659/d309052e-3a68-4768-b4e0-bf76eb43442e)

What often helps me is to set the Macro Security settings as "Disable VBA macros with notification". 
Then when you open the .xll (just before any of your add-in code runs) you get a message asking whether to open or not.
That at least confirms Excel is about to open the add-in.
![image](https://github.com/Excel-DNA/ExcelDna/assets/414659/3a207ade-1ff4-4787-8e81-e2f641b1a04f)
![image](https://github.com/Excel-DNA/ExcelDna/assets/414659/0982b45b-8937-429c-888c-281191f8c1ff)

Otherwise, I would start with a very simple .xll, check whether that works (if not go through security settings etc.). Then add ribbon and check. 

If only ribbon does not work, the one snag I've seen is with some McAfee versions that block the type of COM registry entries we make.
This can be tricky to debug.

**Comment by hxhgxy** [comment]
Thank you very much! 
What I tried is a xll with functions and CTP etc. I will check and try what you guide above later.
